src/discussions-trigger.ts

```diff
@@ -0,0 +1,175 @@
+import { createMutation, client } from "./graphql-client";
+import fetch from "node-fetch";
+import { gql } from "@apollo/client/core";
+import * as crypto from "crypto";
+
+const reply = (context: Context, status: number, body: string) => {
+    context.res = { status, body };
+    context.log.info(`${body} [${status}]`);
+};
+
+export async function run(context: Context, req: HttpRequest) {
+    const isDev = process.env.AZURE_FUNCTIONS_ENVIRONMENT === "Development";
+    const secret = process.env.GITHUB_WEBHOOK_SECRET;
+    const { headers, body } = req;
+    const githubId = headers["x-github-delivery"];
+    const event = headers["x-github-event"]!;
+    const action = body.action;
+
+    context.log(
+    `>>> HTTP Trigger [${event}.${action}; gh: ${githubId}; az: ${context.invocationId}; node: ${process.version}]`
+    );
+
+    // For process.env.GITHUB_WEBHOOK_SECRET see
+    // https://ms.portal.azure.com/#blade/WebsitesExtension/FunctionsIFrameBlade/id/%2Fsubscriptions%2F57bfeeed-c34a-4ffd-a06b-ccff27ac91b8%2FresourceGroups%2Fdtmergebot%2Fproviders%2FMicrosoft.Web%2Fsites%2FDTMergeBot
+    if (!isDev && !(await verify(secret!, body, headers["x-hub-signature-256"]!)))
+        return reply(context, 500, "This webhook did not come from GitHub");
+
+    if (!canHandleRequest(event, action))
+        return reply(context, 204, "Can't handle this request");
+
+    return handleTrigger({ event, action, body }, context);
+}
+
+export const canHandleRequest = (event: string, action: string) => {
+    const name = "discussion";
+    const actions = ["created", "edited"];
+    return event == name && actions.includes(action);
+};
+
+const handleTrigger = (info: { event: string; action: string; body: DiscussionWebhook }, context: Context) => {
+    const categoryID = info.body.discussion.category.slug;
+    if (categoryID === "issues-with-a-types-package") {
+        return pingAuthorsAndSetUpDiscussion(info.body.discussion);
+    } else if (categoryID === "request-a-new-types-package" && info.action === "created") {
+        return updateDiscordWithRequest(info.body.discussion);
+    }
+
+    return reply(context, 204, "Can't handle this specific request");
+};
+
+function extractNPMReference(discussion: Discussion) {
+    const title = discussion.title;
+    if (title.includes("[") && title.includes("]")) {
+        return title.split("[")[1]!.split("]")[0];
+    }
+    return undefined;
+}
+
+const couldNotFindMessage = `Hi, we could not find a reference to the types you are talking about in this discussion. 
+Please edit the title to include the name on npm inside square brackets. 
+
+E.g. 
+ - \`"[@typescript/vfs] Does not x, y"\`
+ - \`"Missing x inside [node]"\`
+ - \`"[express] Broken support for template types"\`
+`;
+
+const gotAReferenceMessage = (module: string, owners: string[]) => `Thanks for the discussion about "${module}", some useful links for everyone: [npm](https://www.npmjs.com/package/${module}) / etc
```

Comment:
(And this would reuse the bits from `compute-pr-actions.ts`, I'm guessing, possibly the popularity labels too.)